### DISCUSSION TITLE
redirect_uri in authorization request

### DISCUSSION THREAD

**Discussion by andrewrmoreland** [keycloak/keycloak#15756] [Q&A] [answered]
An application we are testing doesn't specific the redirect_uri parameter in OIDC authorization requests. My understanding from the standard is that this parameter is optional, and uses the redirect specified in client configuration on the IdP to redirect.

However when using Keycloak as the IdP we get 'Invalid parameter: redirect_uri' error, despite the parameter not being send in the request, and the client configuration in Keycloak containing it.

Is there a configuration option I'm missing for Keycloak? Is the parameter mandatory for Keycloak implementation?

**Comment by sschu** [comment, marked as answer]
The parameter is required in OIDC, see https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest.

**Reply by andrewrmoreland** [reply]
Thanks for that source and clarifying. I had incorrectly read the RFC for OAuth 2.0 rather than OIDC.
